.github/workflows/cd.yml

```
@@ -0,0 +1,100 @@
+name: replicate-cd
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  push:
+    runs-on: ubuntu-latest
+    env:
+      BASE_MODEL: 'replicate-internal/official-sdxl-prod'
+      PROD_MODEL: 'replicate-internal/official-sdxl-prod'
+      PROD_DEPLOYMENT: 'replicate-internal/sdxl-prod-official-deployment'
+
+    steps:
+    - name: checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: what changed
+      id: what-changed
+      run: |
+        FILES_CHANGED=$(git diff --name-only --diff-filter=AMR origin/main HEAD  | xargs)
+        echo "FILES_CHANGED=$FILES_CHANGED" >> $GITHUB_ENV
+        if echo "$FILES_CHANGED" | grep -q 'cog.yaml'; then
+            echo "cog-push=true" >> $GITHUB_OUTPUT
+          else
+            echo "cog-push=false" >> $GITHUB_OUTPUT
+          fi
+        if ${{ contains(github.event.head_commit.message, '[cog build]') }}; then
+            echo "cog-push=true" >> $GITHUB_OUTPUT
+          fi
+
+    # if cog.yaml changes - cog build and push. else - yolo build and push!
+    - name: did-it-tho
+      env:
+        COG_PUSH: ${{ steps.what-changed.outputs.cog-push }}
+      run: |
+        echo "cog push?: $COG_PUSH"
+        echo "changed files: $FILES_CHANGED"
+
+    - name: setup-cog
+      if: steps.what-changed.outputs.cog-push == 'true'
+      uses: replicate/setup-cog@v1.0.3
+      with:
+        token: ${{ secrets.REPLICATE_API_TOKEN }}
+        install-cuda: false
```

```suggestion
      uses: replicate/setup-cog@v2
      with:
        token: ${{ secrets.REPLICATE_API_TOKEN }}
```

Bumped to v2 here, where CUDA drivers are not installed by default.